File: lib/CommandHandlers.ex
defmodule CommandHadlers do
	
	def handle(app, cmd) do
		Repo.get(cmd.id) |>
			domain_event(cmd) |>
			Repo.save(cmd.id, cmd.version, app)
	end
	
	def domain_event(task, %Command{ type: :createTask, title: title }) do
		IO.puts "========= ::createTask"
		DomainTask.create(task, title)
	end
	
	def domain_event(task, %Command{ type: :renameTask, title: title }) do
		IO.puts "========= ::renameTask"
		DomainTask.rename(task, title)	
	end
	
	def domain_event(task, %Command{ type: :completeTask }) do
		IO.puts "========= ::completeTask"
		DomainTask.complete(task)
	end
	
end

File: commandhandlers.js
var repo;

module.exports = {
	create : function(r) {
		repo = r;
		
		return this;
	},
	
	createTask : function (cmd) {
		
		var task = repo.get(cmd.id);

		repo.save(cmd.id, task.create(cmd.id, cmd.title), cmd.version);
	},
		
	renameTask : function (cmd) {
		
		var task = repo.get(cmd.id);
	
		repo.save(cmd.id, task.rename(cmd.title), cmd.version);
	},
		
	completeTask : function (cmd) {
		
		var task = repo.get(cmd.id);
	
		repo.save(cmd.id, task.complete(), cmd.version);
	}
}